.github/workflows/build.yaml

```diff
@@ -0,0 +1,78 @@
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  release:
+    types:
+    - created
+
+env:
+  MASTER_BRANCH: master
+  BUILD_HARNESS_VERSION: 0.5.5
+  HUGO_CONFIG: deploy.toml
+  HTMLTEST_CONFIG: .htmltest.codefresh.yml
+  #TMPDIR: /codefresh/volume
+  YARN_BUILD_DISABLED: false
+  COMPONENTS_BUILD: true
+
+jobs:
+  build-hugo:
+    name: "Build Hugo"
+    runs-on: ubuntu-latest
+    steps:
+    - name: "Checkout source code at current commit"
+      uses: actions/checkout@v2
+    - name: "Set up Docker Buildx"
+      uses: docker/setup-buildx-action@v1
+    - name: "Build docker image"
+      uses: docker/build-push-action@v2
+      with:
+        tags: cloudposse/docs
+        outputs: type=docker,dest=/tmp/image.tar
+    - name: "Import Image"
+      run: |
+        docker load --input /tmp/image.tar
+    - name: "Build Hugo"
+      run: |
+        set +e -x
+        make lint
+        make release
+        make real-clean hugo/build
+    - name: "Verify that Hugo files were built"
+      run: |
+        ls -lhatr
+        ls -lhatr ./public
+
+  hydrate-cache:
+    name: "Hydrate Cache"
+    runs-on: ubuntu-latest
+    steps:
+    - name: "Checkout source code at current commit"
+      uses: actions/checkout@v2
+    - name: "Test Presence of Cache Artifact"
+      id: artifact_test
+      run: |
+        curl -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/cloudposse/docs/actions/artifacts > artifacts.json
+        cat artifacts.json
+        jq --version
+        jq '[.["artifacts"][] | {name: .name, created: .created_at} | select(.name=="htmltest-cache")] | length' artifacts.json > number_of_caches.txt
+        export number_of_caches=`cat number_of_caches.txt`
+        if [ ${number_of_caches} > 0 ]; then
+            echo "::set-output name=cache-exists::true"
+            curl -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/cloudposse/docs/actions/artifacts | jq '[.["artifacts"][] | select(.name=="htmltest-cache")] | sort_by(.created_at) | reverse[0].archive_download_url' > artifact_url.txt
+            export artifact_url=`cat artifact_url.txt`
+            echo "::set-output name=cache-url::${artifact_url}"
+        fi
+    - name: "Download Hugo Site Artifact, If Existent"
+      if:  ${{ steps.artifact_test.outputs.cache-exists }}
+      run: |
+        ls -hratl
+        curl -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" ${{ steps.artifact_test.outputs.cache-url }}
+        ls -hratl
+    - name: "(Re-)Generate cache"
+      run: |
+        make real-clean smoketest
+    - name: "Upload Hugo Site Artifact"
+      uses: actions/upload-artifact@v2
+      with:
+        name: htmltest-cache
+        path: ./public
```

Comment:
Actually, we don't want to cache this. We want to cache the `.htmltest` folder or whatever `OutputDir` is set to.